### PR TITLE
fix: prohibit read commands during takeover

### PR DESCRIPTION
**Pull request by BorysTheDev**
fixes: first item in #4238

### CODE REVIEW

File: src/server/main_service.cc [hunk 1]
```diff
@@ -1057,7 +1057,10 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
       allowed_by_state = false;
       break;
     case GlobalState::TAKEN_OVER:
-      allowed_by_state = !cid->IsWriteOnly();
+      // Only Admin commands are allowed
+      // we prohibit even read commands, because read commands running in pipeline can take a while
+      // to send all data to a client which leads to fail in takeover
+      allowed_by_state = dfly_cntx.conn()->IsPrivileged();
```
Comment (adiholden) [inline src/server/main_service.cc:1063]: if we do this change this means that we can not takeover if replica is not connected on admin port?
is this why you needed all the changes below in replication tests?

Comment (BorysTheDev) [reply src/server/main_service.cc:1063]: yes. And I thought that we do takeover always via admin port

Comment (adiholden) [reply src/server/main_service.cc:1063]: we do not enforce the repltakeover commands on admin port 
I am not sure about this change now as this is a breaking change
what exactly is failing ? is it the ack sent from replica or something else?

Comment (BorysTheDev) [reply src/server/main_service.cc:1063]: current failed test don't connect with replication

File: src/server/main_service.cc [hunk 2]
```diff
@@ -1057,7 +1057,10 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
       allowed_by_state = false;
       break;
     case GlobalState::TAKEN_OVER:
-      allowed_by_state = !cid->IsWriteOnly();
+      // Only Admin commands are allowed
+      // we prohibit even read commands, because read commands running in pipeline can take a while
+      // to send all data to a client which leads to fail in takeover
+      allowed_by_state = dfly_cntx.conn()->IsPrivileged() || cid->name() == "DFLY";
```
Comment (adiholden) [inline src/server/main_service.cc:1063]: DFLY is admin command
CO::ADMIN

you write in comment only admin commands are allowed but the check dfly_cntx.conn()->IsPrivileged() is if the command is from admin port

So maybe just check command flag CO::ADMIN

Comment (BorysTheDev) [reply src/server/main_service.cc:1063]: Not so simple, but it's a good idea I will use it

**Review by adiholden — APPROVED** [review]

**Pull request merged**